pkg/tasks/alertmanager.go

```diff
@@ -117,6 +117,17 @@ func (t *AlertmanagerTask) Run() error {
 	if err != nil {
 		return errors.Wrap(err, "reconciling Alertmanager Service failed")
 	}
+
+	pdb, err := t.factory.PrometheusUserWorkloadPodDisruptionBudget()
+	if err != nil {
+		return errors.Wrap(err, "initializing Alertmanager PodDisruptionBudget object failed")
+	}
+
+	err = t.client.CreateOrUpdatePodDisruptionBudget(pdb)
```

Comment:
pdb may be nil in case of non-HA infrastructure, so you will need to add a check before creating it.